uefi/src/helpers/aligned_buffer.rs

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use alloc::alloc::{alloc, dealloc, Layout, LayoutError};
+use core::error::Error;
+use core::fmt;
+
+/// Helper class to maintain the lifetime of a memory region allocated with a non-standard alignment.
+/// Facilitates RAII to properly deallocate when lifetime of the object ends.
+///
+/// Note: This uses the global Rust allocator under the hood.
+#[allow(clippy::len_without_is_empty)]
+#[derive(Debug)]
+pub struct AlignedBuffer {
```

Comment:
I'm fine with keeping this struct. However, the `helpers` module is supposed to provide  helpers to interact with the outer environment.

Therefore, I see better fit for `AlignedBuffer` in `<uefi>/mem/mod.rs`.

Are you fine with keeping the AlignedBuffer struct, @nicholasbishop ?